### ISSUE TITLE
Keyboard layout ignored for keybinds

### ISSUE THREAD

**Issue by zeroeightysix** [ocornut/imgui#2959]
**Version/Branch of Dear ImGui:**

Version: 1.75 WIP (latest commit)
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Operating System: Linux (Arch)
Keyboard layout: AZERTY

> Note: Testing was done using the **example_glfw_opengl3** example

**My Issue:**

Text entry widgets ignore keyboard layout: all keybinds, e.g. *Select all (<kbd>CTRL</kbd>+<kbd>A</kbd>)* assume the user is using QWERTY.
Someone typing using AZERTY (France, Belgium) or any **non-qwerty layout** will have to press the keys where <kbd>A</kbd> is on QWERTY, instead of pressing the actual <kbd>A</kbd> key.
For AZERTY, <kbd>Q</kbd> is in the same place as the <kbd>A</kbd> key on QWERTY, making it so that <kbd>CTRL</kbd>+<kbd>Q</kbd> will result in a "Select all" operation happening.

This will happen with other keybinds, such as <kbd>CTRL</kbd>+<kbd>Z</kbd> etc.

**Screenshots/Video**

[Video](https://giphy.com/gifs/MAWDWawv0afgGLEE2a/html5)

First I type all `a`s to show the key code of `a`, and then attempt to press <kbd>CTRL</kbd>+<kbd>A</kbd>.
After that, I do the same for <kbd>Q</kbd>. Only then a selection happens.

**Standalone, minimal, complete and verifiable example:**

[examples/example_glfw_opengl3](https://github.com/ocornut/imgui/tree/master/examples/example_glfw_opengl3)
All can be tested in the demo window(s).

**Comment by ocornut** [comment]
Hello @zeroeightysix and thank you for your report.

You are right about this. Similar things are discussed in #2625 and #456.
In particular see https://github.com/ocornut/imgui/pull/2625#issuecomment-502091198 then the discussion in https://github.com/glfw/glfw/issues/672#issuecomment-472803131 (messages from March 2019) then forked into https://github.com/glfw/glfw/issues/1502

My opinion is that GLFW doesn't provide the info we need in an ideal manner. Somehow the ideas discussed in glfw's 672 derailed a little in 1502 into api details. 

Realistically speaking we should:
- Figure out a workaround with current GLFW version (perhaps using `glfwGetKeyName()`)
- Suggest/design/push for a correcter fix in GLFW via a new API (that's the purpose of the 1502).
- May be some redesign needed on dear imgui side.

TL;DR; not a trivial issue to move forward on, but we will eventually.

**Comment by blep** [comment]
For SDL2, I solved this by first initializing the scancodes to unknown, and then dynamically setting them on keypress looking at the ASCII code of the pressed letter.

Here a [gist](https://gist.github.com/blep/59b5a69b9671f0a90f3576d14520eb5c) with the change I made. Not having `Ctrl+A` and `Ctrl+Z` working as usual in text field was driving me nuts.

Not clear to me if GLFW provides a way to get the ASCII code associated to a keypress that would allow this "trick" to be used. SDL2 had a function to statically returns the letter associated to a scancode, but it was hard-coded to QWERTY (my first failed attempt at a solution for this problem).

**Comment by keur** [comment]
Also on Linux and this is happening to me with Enter and Backspace. I set a breakpoint in `ImGui_ImplGlfw_KeyCallback`. `key` is -1 and the scancode corresponds to the value from `xev`.

**Comment by jamesthomasgriffin** [comment]
I also encountered this problem when using a Dvorak layout.  I used glfwGetKeyName to remap the keys in the callback, see below.  Ofcourse this does mean the connection is lost between the GLFW_KEY_ names and the physical keys (assuming a US layout), so for example this isn't appropriate for WASD arrow keys.  But it was what I needed.

I'm not sure how it would be possible to integrate this into Dear ImGui (perhaps an option in imconfig.h?), but in the future having both IsKeyPressed(int) for physical keys and IsKeyWithNamePressed(char) or similar would be convenient.

```  cpp
void ImGui_ImplGlfw_KeyCallback(GLFWwindow* window, int key, int scancode, int action, int mods)
  {
    if (g_PrevUserCallbackKey != NULL)
        g_PrevUserCallbackKey(window, key, scancode, action, mods);

    // New code - remap keys using glfwGetKeyName
    const char* key_name = glfwGetKeyName(key, 0);
    if (key_name)
    {
        key = (int)key_name[0];

        // Convert to upper-case
        if (key >= 97 && key <= 122)
            key -= 32;  
    }

    ImGuiIO& io = ImGui::GetIO();
    if (action == GLFW_PRESS)
        io.KeysDown[key] = true;
    if (action == GLFW_RELEASE)
        io.KeysDown[key] = false;

    // Modifiers are not reliable across systems
    io.KeyCtrl = io.KeysDown[GLFW_KEY_LEFT_CONTROL] || io.KeysDown[GLFW_KEY_RIGHT_CONTROL];
    io.KeyShift = io.KeysDown[GLFW_KEY_LEFT_SHIFT] || io.KeysDown[GLFW_KEY_RIGHT_SHIFT];
    io.KeyAlt = io.KeysDown[GLFW_KEY_LEFT_ALT] || io.KeysDown[GLFW_KEY_RIGHT_ALT];
#ifdef _WIN32
    io.KeySuper = false;
#else
    io.KeySuper = io.KeysDown[GLFW_KEY_LEFT_SUPER] || io.KeysDown[GLFW_KEY_RIGHT_SUPER];
#endif
}
```

p.s. this is my first comment in this repo, so a huge thanks to ocornut and anyone else for doing such a great job

**Comment by ocornut** [comment]
Hello,
Stumbled on this through a link that got shared to me and noticed that this should have been closed a while ago.
#4921 (~january 2022) standardized the fact that backends submit translated keys so this should be solved already!
